src/Checkout.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Laravel\Cashier;
+
+use Illuminate\Support\Facades\View;
+use Stripe\Checkout\Session;
+
+class Checkout
+{
+    /**
+     * The Stripe model instance.
+     *
+     * @var \Illuminate\Database\Eloquent\Model
+     */
+    protected $owner;
+
+    /**
+     * The Stripe Checkout Session instance.
+     *
+     * @var \Stripe\Checkout\Session
+     */
+    protected $session;
+
+    /**
+     * Create a new Checkout Session instance.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $owner
+     * @param  \Stripe\Checkout\Session  $session
+     * @return void
+     */
+    public function __construct($owner, Session $session)
+    {
+        $this->owner = $owner;
+        $this->session = $session;
+    }
+
+    /**
+     * Begin a new Checkout Session.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $owner
+     * @param  array  $sessionOptions
+     * @param  array  $customerOptions
+     * @return \Laravel\Cashier\Checkout
+     */
+    public static function create($owner, array $sessionOptions = [], array $customerOptions = [])
+    {
+        $customer = $owner->createOrGetStripeCustomer($customerOptions);
+
+        $session = Session::create(array_merge([
+            'customer' => $customer->id,
+            'mode' => 'payment',
+            'success_url' => $sessionOptions['success_url'] ?? route('home').'?checkout=success',
+            'cancel_url' => $sessionOptions['cancel_url'] ?? route('home').'?checkout=cancelled',
+            'payment_method_types' => ['card'],
+        ], $sessionOptions), Cashier::stripeOptions());
+
+        return new static($customer, $session);
+    }
+
+    /**
+     * Get the View instance for the button.
+     *
+     * @param  string  $label
+     * @param  array  $options
+     * @return \Illuminate\Contracts\View\View
+     */
+    public function button($label = 'Check out', array $options = [])
```

Comment:
nitpick but it seems like everywhere it is referred to as `Checkout` not `Check out`

Reply:
https://twitter.com/stauffermatt/status/1036953677704163330